### DISCUSSION TITLE
Style list item bullets/numbers

### DISCUSSION THREAD

**Discussion by gjblajian** [dolanmiu/docx#2046] [Q&A]
@dolanmiu How would I style the bullet or number for a list item (or set of list items) with respect to size, italics, bold, underline, and font. Since each list can have a different set of styles it does not make sense to style them all in the numbering section of the IPropertiesOptions. Seems like you would need to set them on the individual list item paragraphs (even though these specific properties are TextRun level not Paragraph level). Do you have an example of styling the bullets / numbers on list items?

**Comment by dolanmiu** [comment]
You can add styles to bullet or list items by setting the `style` property:

https://github.com/dolanmiu/docx/blob/master/src/file/numbering/level.ts#L209-L212

Which has the options:

https://github.com/dolanmiu/docx/blob/master/src/file/paragraph/run/properties.ts#L36-L76

An example of it can be found here:

https://github.com/dolanmiu/docx/blob/master/demo/3-numbering-and-bullet-points.ts#L17-L21

Is this what you're looking for?

**Reply by gjblajian** [reply]
Not quite, I am generating the paragraphs at run time and do not necessarily know what the overall document styling is since I am iterating over input elements to produce the docx objects. What I am hoping to find out is if there is a way to set the styles on the individual paragraphs to which the numbering styles apply, vice the document style. Or am I misunderstanding something here?

**Reply by dolanmiu** [reply]
I see, it is possible to set the styles on a `TextRun` level during runtime, like so:

https://github.com/dolanmiu/docx/blob/master/demo/1-basic.ts#L16-L17

Full list of options is here:

https://github.com/dolanmiu/docx/blob/master/src/file/paragraph/run/properties.ts#L37-L75

**Reply by gjblajian** [reply]
@dolanmiu I think I see how to do this now. We have to create our own set of list references and custom levels as we are iterating over the nodes, styling their textRun properties as we go, and collect them all to insert into the numbering property of the IPropertiesOptions at the end when we actually create the Document object. 

![image](https://user-images.githubusercontent.com/5183939/228373572-e249906f-b948-4ae7-abe3-cd554bae323e.png)

document.xml
```
    <w:p w14:paraId="619B20B5" w14:textId="004CF821" w:rsidR="008E6384" w:rsidRDefault="008E6384" w:rsidP="008E6384">
      <w:pPr>
        <w:pStyle w:val="ListParagraph"/>
        <w:numPr>
          <w:ilvl w:val="0"/>
          <w:numId w:val="4"/>
        </w:numPr>
      </w:pPr>
      <w:r>
        <w:t>Item A</w:t>
      </w:r>
    </w:p>
```

numbering.xml
```
  <w:abstractNum w:abstractNumId="1" w15:restartNumberingAfterBreak="0">
    <w:nsid w:val="31AB3171"/>
    <w:multiLevelType w:val="hybridMultilevel"/>
    <w:tmpl w:val="F5462BB6"/>
    <w:lvl w:ilvl="0" w:tplc="6EF659AC">
      <w:start w:val="1"/>
      <w:numFmt w:val="upperLetter"/>
      <w:lvlText w:val="%1."/>
      <w:lvlJc w:val="left"/>
      <w:pPr>
        <w:ind w:left="720" w:hanging="360"/>
      </w:pPr>
      <w:rPr>
        <w:b/>
        <w:i/>
        <w:sz w:val="28"/>
      </w:rPr>
    </w:lvl>
...
  <w:num w:numId="4" w16cid:durableId="163403500">
    <w:abstractNumId w:val="1"/>
  </w:num>
...
```

I was hoping there might be a less "global" method of accomplishing this but it would appear that is not really the case. Word really only understands the list properties based on the styling defined in numbering.xml. Thank you for your assistance.
